src/common/internal/tree.ts

```diff
@@ -185,16 +205,45 @@ export class TestTree {
   }
 
   /** Propagate the subtreeTODOs/subtreeTests state upward from leaves to parent nodes. */
-  static propagateCounts(subtree: TestSubtree): { tests: number; nodesWithTODO: number } {
-    subtree.subtreeCounts ??= { tests: 0, nodesWithTODO: 0 };
+  static async propagateCounts(
+    subtree: TestSubtree,
+    chunking: { metadata: TestMetadataListing; maxChunkTime: number } | undefined
+  ): Promise<{ tests: number; nodesWithTODO: number; totalTimeMS: number; subcaseCount: number }> {
+    subtree.subtreeCounts ??= { tests: 0, nodesWithTODO: 0, totalTimeMS: 0 };
+    subtree.subcaseCount = 0;
     for (const [, child] of subtree.children) {
       if ('children' in child) {
-        const counts = TestTree.propagateCounts(child);
+        const counts = await TestTree.propagateCounts(child, chunking);
         subtree.subtreeCounts.tests += counts.tests;
         subtree.subtreeCounts.nodesWithTODO += counts.nodesWithTODO;
+        subtree.subtreeCounts.totalTimeMS += counts.totalTimeMS;
+        subtree.subcaseCount += counts.subcaseCount;
+      } else {
+        subtree.subcaseCount = child.subcaseCount;
       }
     }
-    return subtree.subtreeCounts;
+
+    if (chunking && subtree.query instanceof TestQueryMultiCase) {
```

Comment:
sanity check: chunking will mean we do/don't split at the test level. so in the end you could a mix of any:
 - TestQuerySingleCase meaning we broke up all the cases like `s:f:t:foo,abc`, `s:f:t:foo,def`
 - TestQueryMultiCase where it is all cases for a test like `s:f:t:*`
 - TestQueryMultiCase where some of the cases are split - like `s:f:t:foo,*` and `s:f:t:bar,*`
 - mix of single/multi like `s:f:t:foo,abc`, `s:f:t:foo,def`, `s:f:t:bar,*`, `s:f:t:bar,*`
 
but it'll never merge up to `s:f:t1,*` or `s:f:*` ?

Reply:
correct.

    // - TestQueryMultiTest and higher nodes are never set to collapsible anyway, so we ignore them.
    // - TestQuerySingleCase nodes can't be collapsed, so we ignore them.